templates/demo-store/tsconfig.json

```diff
@@ -5,7 +5,7 @@
     "isolatedModules": true,
     "esModuleInterop": true,
     "jsx": "react-jsx",
-    "moduleResolution": "node",
+    "moduleResolution": "Bundler",
```

Comment:
This [new resolution mode](https://devblogs.microsoft.com/typescript/announcing-typescript-5-0/#moduleresolution-bundler) is supposed to be appropriate for ESBuild/etc.